plugins/samples/domain_rewrite/tests.textpb

```diff
@@ -0,0 +1,40 @@
+test {
+  name: "DomainRewrite"
+  num_chunks: 6
+  response_body {
+    input { content:  "<!DOCTYPE html>"
+                      "<html>"
+                      "<head>"
+                      "<title>Page Title</title>"
+                      "</head>"
+                      "<body>"
+
+                      "<h1>This is a Heading</h1>"
+                      "<p>This is a paragraph.</p>"
+                      "<div>"
+                      "<a href=https://foo.com>website</a>"
+                      "</div>"
+                      "<a href=https://foo.com>website</a>"
+                      "</body>"
+                      "</html>"
+    }
+    result {
+      body { exact: "<!DOCTYPE html>"
+                    "<html>"
+                    "<head>"
+                    "<title>Page Title</title>"
+                    "</head>"
+                    "<body>"
+
+                    "<h1>This is a Heading</h1>"
+                    "<p>This is a paragraph.</p>"
+                    "<div>"
+                    "<a href=\"https://bar.com\">website</a>"
+                    "</div>"
+                    "<a href=https://foo.com>website</a>"
+                    "</body>"
+                    "</html>"
+      }
+    }
+  }
+}
```

Comment:
Can you split/add a few more tests:


1. one chunk
1. many chunks
1. a test that some unrelated instance of `foo` outside of an href is not modified
1. split out the test for the second `foo` into a separate test that explicitly tests for unchanged hrefs after the stop condition